flairsou_frontend/src/components/App.jsx

```diff
@@ -46,15 +48,137 @@ PrivateRoute.defaultProps = {
   location: null,
 };
 
-const App = () => (
-  <React.StrictMode>
-    <BrowserRouter>
-      <Switch>
-        <Route path="/" exact component={Home} />
-        <Route path="/account/:accountID" exact component={Account} />
-      </Switch>
-    </BrowserRouter>
-  </React.StrictMode>
-);
+// routes de l'API
+const authlinkUrl = '/oauth/authlink';
+const userInfosUrl = '/proxy_pda/get_user_infos';
+const assosUrl = '/proxy_pda/get_list_assos';
+
+const App = () => {
+  // informations sur l'utilisateur
+  const [user, setUser] = useState({ id: null, firstname: null, lastname: null });
+
+  // liste des assos de l'utilisateur
+  const [assos, setAssos] = useState([]);
+
+  // association active dans l'application
+  const [assoActive, setAssoActive] = useState(null);
+
+  // liste des comptes liés à l'association active dans l'application
+  const [accountList, setAccountList] = useState([]);
+
+  // clé primaire du compte actif
+  const [accountActive, setAccountActive] = useState(-1);
+
+  // fonction permettant de récupérer la clé primaire de chaque livre associé à chaque
+  // association de la liste fournie, puis de mettre à jour l'état de la liste des
+  // assos
+  async function fetchBookPks(assosList) {
+    // fonction de récupération des livres associés aux entités
+    const fetchAssoBook = async (asso) => {
+      // appel de l'API pour récupérer le livre associé à l'asso
+      const response = await fetch(`/api/books/byEntity/${asso.asso_id}/`);
+      let bookId = '-1';
+      if (response.status === 200) {
+        const json = await response.json();
+        bookId = json[0].pk;
+      }
+
+      // récupération récursive des livres pour les sous-assos
+      const subAssos = await Promise.all(
+        asso.asso_set.map((subAsso) => fetchAssoBook(subAsso)),
+      );
+
+      // renvoi du résultat
+      return {
+        ...asso,
+        asso_set: subAssos,
+        book: bookId,
+      };
+    };
+
+    const newAssos = await Promise.all(
+      assosList.map((asso) => fetchAssoBook(asso)),
+    );
+    setAssos(newAssos);
+  }
+
+  // TODO : peut être fetch tous les comptes de toutes les assos pour ne pas fetch une liste
+  // à chaque fois que l'utilisateur clique sur le bouton ? après c'est mis en cache du navigateur
+  // donc bon... à voir
+
+  // récupération des informations de l'utilisateur
+  useEffect(() => {
+    fetch(userInfosUrl)
+      .then((response) => {
+        if (response.status === 403) {
+          // l'utilisateur n'est pas connecté, on le renvoie vers la page de connexion
+          fetch(authlinkUrl)
+            .then((newResponse) => newResponse.json())
+            .then((newResponse) => {
+              window.location.href = newResponse.link;
```

Comment:
`return <Redirect to={newResponse.link} />;`
Non ?

Reply:
Dans la Promise ça a pas l'air de fonctionner, on verra si on change ça plus tard je pense.